src/test/extensions/extensionsHelper.spec.ts

```diff
@@ -977,6 +895,108 @@ describe("extensionsHelper", () => {
     });
   });
 
+  describe("getNextVersionByStage", () => {
+    let listExtensionVersionsStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      listExtensionVersionsStub = sinon.stub(publisherApi, "listExtensionVersions");
+    });
+
+    afterEach(() => {
+      listExtensionVersionsStub.restore();
+    });
+
+    it("should return expected stages and versions", async () => {
+      listExtensionVersionsStub.returns(
+        Promise.resolve([
+          { spec: { version: "1.0.0-rc.0" } },
+          { spec: { version: "1.0.0-rc.1" } },
+          { spec: { version: "1.0.0-beta.0" } },
+        ])
+      );
+      const expected = new Map<string, string>([
+        ["rc", "1.0.0-rc.2"],
+        ["alpha", "1.0.0-alpha.0"],
+        ["beta", "1.0.0-beta.1"],
+        ["stable", "1.0.0"],
+      ]);
+      const { versionByStage, hasVersions } = await extensionsHelper.getNextVersionByStage(
+        "test",
+        "1.0.0"
+      );
+      expect(Array.from(versionByStage.entries())).to.eql(Array.from(expected.entries()));
+      expect(hasVersions).to.eql(true);
+    });
+
+    it("should ignore unknown stages and different prerelease format", async () => {
+      listExtensionVersionsStub.returns(
+        Promise.resolve([
+          { spec: { version: "1.0.0-beta" } },
+          { spec: { version: "1.0.0-prealpha.0" } },
+        ])
+      );
+      const expected = new Map<string, string>([
+        ["rc", "1.0.0-rc.0"],
+        ["alpha", "1.0.0-alpha.0"],
+        ["beta", "1.0.0-beta.0"],
+        ["stable", "1.0.0"],
+      ]);
+      const { versionByStage, hasVersions } = await extensionsHelper.getNextVersionByStage(
+        "test",
+        "1.0.0"
+      );
+      expect(Array.from(versionByStage.entries())).to.eql(Array.from(expected.entries()));
+      expect(hasVersions).to.eql(true);
+    });
+  });
+
+  describe("unpackExtensionState", () => {
+    it("should return correct state", () => {
```

Comment:
Better to separate these into multiple it blocks, one for each call to unpackExtensionState. That way, they will fail in a more granular manner